src/secret/ledger_nano.rs

```diff
@@ -95,160 +114,217 @@ impl SecretManage for LedgerSecretManager {
-                            // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
-                            .map(|seg| u32::from_be_bytes(seg.bs()))
-                            .collect();
-                        (raw[0], raw[1])
+        let mut input_bip32_indices: Vec<LedgerBIP32Index> = Vec::new();
+        let mut coin_type: Option<u32> = None;
+        let mut account_index: Option<u32> = None;
+
+        let input_len = prepared_transaction.inputs_data.len();
+
+        for input in &prepared_transaction.inputs_data {
+            let bip32_indices: Vec<u32> = match &input.chain {
+                Some(chain) => {
+                    chain
+                        .segments()
+                        .iter()
+                        // XXX: "ser32(i)". RTFSC: [crypto::keys::slip10::Segment::from_u32()]
+                        .map(|seg| u32::from_be_bytes(seg.bs()))
+                        .collect()
+                }
+                None => return Err(crate::Error::NoInputs),
+            };
+            // coin_type and account_index should be the same in each output
+            if (coin_type.is_some() && coin_type != Some(bip32_indices[0]))
+                || (account_index.is_some() && account_index != Some(bip32_indices[1]))
+            {
+                return Err(crate::Error::InvalidBIP32ChainData);
+            }
+
+            coin_type = Some(bip32_indices[1]);
+            account_index = Some(bip32_indices[2]);
+            input_bip32_indices.push(iota_ledger::LedgerBIP32Index {
+                bip32_change: bip32_indices[3] | HARDENED,
+                bip32_index: bip32_indices[4] | HARDENED,
+            });
+        }
+
+        if coin_type.is_none() || account_index.is_none() {
+            return Err(crate::Error::NoInputs);
+        }
+
+        // unwrap values
+        let coin_type = coin_type.unwrap() & !HARDENED;
+        let bip32_account = account_index.unwrap() | HARDENED;
+
+        // pack essence and hash into vec
+        let essence_bytes = prepared_transaction.essence.pack_to_vec();
+        let essence_hash = prepared_transaction.essence.hash().to_vec();
+
+        let ledger = iota_ledger::get_ledger(coin_type, bip32_account, self.is_simulator)?;
+        let blind_signing = needs_blind_signing(prepared_transaction, ledger.get_buffer_size());
+
+        // if essence + bip32 input indices are larger than the buffer size or the essence contains
+        // features / types that are not supported blind signing will be needed
+        if blind_signing {
+            // prepare signing
+            log::debug!("[LEDGER] prepare_blind_signing");
+            log::debug!("[LEDGER] {:?} {:?}", input_bip32_indices, essence_hash);
+            ledger.prepare_blind_signing(input_bip32_indices, essence_hash)?;
+        } else {
+            // figure out the remainder address and bip32 index (if there is one)
+            let (has_remainder, remainder_address, remainder_bip32): (
```

Comment:
`has_remainder` is useless since `remainder_address` is an Option

Reply:
k, removed :blush: